pkg/gce-pd-csi-driver/utils.go

```diff
@@ -63,7 +64,12 @@ func logGRPC(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, h
 	// Note that secrets are not included in any RPC message. In the past protosanitizer and other log
 	// stripping was shown to cause a significant increase of CPU usage (see
 	// https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/356#issuecomment-550529004).
-	klog.V(4).Infof("%s called with request: %s", info.FullMethod, req)
+	// However malicious user still can put a secret in request as explained here:
+	// https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/1372
+	if klog.V(4).Enabled() {
```

Comment:
We regularly use log level 4 in production. Since this driver doesn't actually require any secret, can we just clear the secrets from the request?

Reply:
@msau42 , the request may be of different types, i.e. `NodeExpandVolumeRequest` is not the only type with `Secrets` (e.g. `CreateVolumeRequest`, `DeleteVolumeRequest`, etc.). I didn't find any straightforward way to clear the secrets from a request of arbitrary type, can you tell me how you suggest to do it?

Reply:
Do you think we can leverage the reflect package? https://pkg.go.dev/reflect#example-Value.FieldByName potentially looks promising. Or could we even use https://pkg.go.dev/reflect#StructTag to create a more efficient protosanitizer?